### PR TITLE
Калистратов - Лабораторная работа #1

### PR DESCRIPTION
**Pull request by Andreyka452**
@ErmolaevMakcum @samarkinav

### CODE REVIEW

File: code/game_of_life/game_of_life.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+                        raise ValueError(msg)
+        return next_field
+
+    def string_validation(self, field_lines):
+        for field_line in field_lines:
+            if len(field_lines[0]) != len(field_line):
+                msg = "invalid string entered"
+                raise ValueError(msg)
+
+    def calc_filed_boundaries(self, field_lines):
```
Comment (ErmolaevMakcum) [inline code/game_of_life/game_of_life.py:108]: typo in the method name "filed" -> "field"

File: code/game_of_life/game_of_life.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+                if field_lines[i][j] == '.':
+                    next_field[i][j] = 0
+                elif field_lines[i][j] == '*':
+                    next_field[i][j] = 1
+                else:
+                        msg = "invalid character entered" + ', ' + field_lines[i][j]
+                        raise ValueError(msg)
+        return next_field
+
+    def string_validation(self, field_lines):
```
Comment (ErmolaevMakcum) [inline code/game_of_life/game_of_life.py:102]: Unused method

File: code/game_of_life/game_of_life.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        for field_line in field_lines:
+            if len(field_lines[0]) != len(field_line):
+                msg = "invalid string entered"
+                raise ValueError(msg)
```
Comment (ErmolaevMakcum) [inline code/game_of_life/game_of_life.py:106]: Unused method

File: code/game_of_life/game_of_life.py [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+        next_field = [[0 for i in range(field_boundaries["width_field"])]
+                      for j in range(field_boundaries["height_field"])]
+        return next_field
+
+    def step_calculation(self, current_field, field_boundaries):
+        next_field = self.create_empty_field(field_boundaries)
+        for i in range(field_boundaries["height_field"]):
+            for j in range(field_boundaries["width_field"]):
+                current_neighbors = self.counting_neighbors(current_field, i, j)
+                if current_field[i][j]:
```
Comment (samarkinav) [inline code/game_of_life/game_of_life.py:30]: I think `if current_field[i][j] == 1` looks better. It works in your task, but I think it's better to do this when you have the bool variable.

File: code/game_of_life/game_of_life.py [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+                    if current_neighbors == 2 or current_neighbors == 3:
+                        next_field[i][j] = 1
+                    else:
+                        next_field[i][j] = 0
+                else:
```
Comment (samarkinav) [inline code/game_of_life/game_of_life.py:35]: And here it is better to compare with zero

Comment (samarkinav) [reply code/game_of_life/game_of_life.py:35]: with using "elif"

**Comment by Andreyka452** [comment]
@VikaSannikova

**Comment by samarkinav** [comment]
:+1:

**Comment by ErmolaevMakcum** [comment]
👍

**Comment by Andreyka452** [comment]
@kirill-korniakov

**Pull request merged**